samza-core/src/main/java/org/apache/samza/clustermanager/JobCoordinatorLaunchUtil.java

```diff
@@ -57,14 +58,17 @@ public static void run(SamzaApplication app, Config config) {
       throw new SamzaException("Only support single remote job is supported.");
     }
 
-    Config finalConfig = jobConfigs.get(0);
+    Config fullConfig = jobConfigs.get(0);
+    MetricsRegistryMap metrics = new MetricsRegistryMap();
+    MetadataStore
```

Comment:
nit pick: new CoordinatorStreamStore(CoordinatorStreamUtil.buildCoordinatorStreamConfig(fullConfig), new MetricsRegistryMap());

Reply:
Cannot do it because the same MetricsRegistryMap needs to be passed in ClusterBasedJobCoordinator's constructor as well.